schedule/views.py

```diff
@@ -8,25 +7,41 @@
 from forecaster.forecaster import forecast as c2alg2 # company 2 alg 2
 
 from c1algo1 import scheduler as c1alg1# company 1 alg 1
-from c1algo2.forecaster import forecast as c1algo2
+from c1algo2.forecaster import forecast as c1alg2
 
+from schedule.alg_data_generator import get_historic_course_data
+from schedule.alg_data_generator import get_program_enrollment_data
+from schedule.alg_data_generator import get_schedule
+from schedule.alg_data_generator import get_professor_dict_mock
+from schedule.alg_data_generator import get_schedule_alg1_mock
+from schedule.alg_data_generator import get_schedule_alg2_mock
+
+import traceback
+import json
+
+from courses.models import Course
 
 
 class Schedule(APIView):
     # GET / schedule / {year - semester}
     def get(self, request: HttpRequest, year: int, semester: str, requested_company_alg: int) -> HttpResponse:
-        alg2_result = c2alg1(None, None, None) if requested_company_alg == 1 else c2alg2(None, None, None)
-        alg1_result = c1alg1.generate_schedule() if requested_company_alg == 1 else c2alg1(None, None, None)
-
-        # these 2 lines are temporary and just to make unit tests pass while integration is in progress
-        alg2_result = "OK"
-        alg1_result = "OK"
-
-        if "OK" == alg1_result and "OK" == alg2_result:
-            body = "GENERATED SCHEDULE FROM COMPANY %d ALGORITHM, AND BOTH RETURNED OK" % requested_company_alg
-            return HttpResponse(body, status=status.HTTP_200_OK)
-        else: 
-            return HttpResponse(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        # Create params for algorithms packages
+        historical_data = get_historic_course_data()
+        previous_enrollment = get_program_enrollment_data()
+        schedule = get_schedule_alg2_mock()
+        professors = get_professor_dict_mock()
+        schedule_1 = get_schedule_alg1_mock()
+
+        try:
+            schedule = c1alg2(historical_data, previous_enrollment, schedule) if requested_company_alg == 1 \
+                 else c2alg2(historical_data, previous_enrollment, None)
```

Comment:
If the mock data you are using for c1alg2 works for their algorithm but you can't use it for c2alg2 then that is something our alg 2 team needs to address before the demo.

We should be using the same mock data for both company's algorithms. It's easiest to have that assurance if we don't use the flag, but if you want to use the flag, then please ensure that `schedule` _can_ be used for our alg 2. If it cannot, then that either means your copy of the mock data is out of date, or company 1 won't be able to call our algorithm 2.